### PR TITLE
chore(query): reduce reallocation for aggregator state builders

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

reduce reallocation for aggregator state builders 

1. For `count`, `sum`, `avg`, `min`, `max` we use fixed-int encoding to align the states
2. Introduce `batch_serialize` API in aggregator function

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13193)

### CODE REVIEW

**Comment by sundy-li** [comment]
askbend:summary

File: src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_final.rs [hunk 1]
```diff
@@ -136,28 +136,19 @@ where Method: HashMethodBounds
...
-                            let aggr_column =
-                                agg.value.as_column().unwrap().as_string().ok_or_else(|| {
-                                    ErrorCode::IllegalDataType(format!(
-                                        "Aggregation column should be StringType, but got {:?}",
-                                        agg.value
-                                    ))
-                                })?;
-                            states_binary_columns.push(aggr_column);
+                            let col = agg.value.as_column().unwrap();
+                            states_binary_columns.push(col.slice(0..places.len()));
```
Comment (RinChanNOWWW) [inline src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_final.rs:140]: `places.len()` should equal to `agg.len()`. We can add an assertion and remove the slice operation.

Comment (sundy-li) [reply src/query/service/src/pipelines/processors/transforms/aggregator/transform_aggregate_final.rs:140]: there is `reach_limit` which could make places size shorter than `agg.len()`

File: src/query/functions/src/aggregates/aggregate_min_max_any.rs [hunk 2]
```diff
@@ -120,14 +126,23 @@ where
     fn serialize(&self, place: StateAddr, writer: &mut Vec<u8>) -> Result<()> {
         let state = place.get::<State>();
 
-        serialize_state(writer, state)
+        if self.serialize_size_per_row().is_some() {
+            serialize_fixed_state(writer, state)
```
Comment (RinChanNOWWW) [inline src/query/functions/src/aggregates/aggregate_min_max_any.rs:130]: How about add a template parameter to indicate if need fixed state.

File: src/query/functions/src/aggregates/aggregate_sum.rs [hunk 3]
```diff
@@ -342,13 +344,22 @@ where State: SumState
 
     fn serialize(&self, place: StateAddr, writer: &mut Vec<u8>) -> Result<()> {
         let state = place.get::<State>();
-        serialize_state(writer, state)
+        if self.serialize_size_per_row().is_some() {
```
Comment (RinChanNOWWW) [inline src/query/functions/src/aggregates/aggregate_sum.rs:347]: Ditto.

**Pull request closed without merging**